### PR TITLE
Issue #303 Account Dialog

### LINKED ISSUE #303: Add Account Dialog with Delete
Add an option to the account menu in upper right to view the account.

This should show the email, name, image, and granted role.

It should have a button to request their account be deleted.

### PR DESCRIPTION
**Pull request by mckoon**
* Splits authorization module into service and component modules.
* Creates "View account" menu option.
* Adds delete account ability to current user.

Closes #303

### CODE REVIEW

File: client/src/main/webjar/authentication/authentication.component.js [hunk 1]
```diff
@@ -1,52 +0,0 @@
-(function (angular) {
```
Comment (mckoon) [inline client/src/main/webjar/authentication/authentication.component.js]: To keep from either having everything in the authentication module, or having circular dependency issues, I split the component out of the authentication module and into a new user-account-actions module.

File: client/src/main/webjar/user-account-actions/user-account-actions.component.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+(function (angular) {
```
Comment (mckoon) [inline client/src/main/webjar/user-account-actions/user-account-actions.component.js]: This file was the authentication component.

File: client/src/main/webjar/user-list/user-list.component.js [hunk 3]
```diff
@@ -26,28 +26,6 @@
...
-
-            } else if (user.admin) {
-                role = "Admin";
-
-            } else {
-                role = "General user";
-            }
-
-            return role;
-        }
```
Comment (mckoon) [inline client/src/main/webjar/user-list/user-list.component.js]: I needed this in the new dialog, so I split it out into a service to share it.

**Comment by mckoon** [comment]
# Screenshots


screen shot 2016-11-27 at 8 25 09 pm



screen shot 2016-11-27 at 8 25 35 pm

**Comment by mckoon** [comment]
https://hmhb-mckoon.herokuapp.com

File: client/src/main/webjar/user-account-details/user-account-details.controller.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        $mdDialog,
+        $log,
+        authenticationService,
+        User,
+        userManagementService
+    ) {
+        var userAccountDetails = this;
+
+        /**
+         * Handles a successful save of the Organization.
```
Comment (RyanBard) [inline client/src/main/webjar/user-account-details/user-account-details.controller.js]: Copy paste error :)

**Pull request merged**